source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs

```
@@ -113,41 +115,34 @@ private static BusinessProcessResult ValidateGridArea(CreateMeteringPoint reques
             return new BusinessProcessResult(request.TransactionId, validationRules);
         }
 
-        private static BusinessProcessResult CheckBusinessRules(CreateMeteringPoint request, Domain.MeteringPoints.Consumption.MeteringPointDetails meteringPointDetails)
+        private static BusinessProcessResult CheckBusinessRules(CreateMeteringPoint request, MeteringPointDetails meteringPointDetails)
         {
+            BusinessRulesValidationResult validationResult;
             var meteringPointType = EnumerationType.FromName<MeteringPointType>(request.TypeOfMeteringPoint);
-            if (meteringPointType != MeteringPointType.Consumption)
+
+            switch (meteringPointType.Name)
             {
-                return BusinessProcessResult.Ok(request.TransactionId);
+                case nameof(MeteringPointType.Consumption):
+                    validationResult = ConsumptionMeteringPoint.CanCreate((ConsumptionMeteringPointDetails)meteringPointDetails);
+                    break;
+                case nameof(MeteringPointType.Production):
+                    validationResult = ProductionMeteringPoint.CanCreate((ProductionMeteringPointDetails)meteringPointDetails);
+                    break;
+                default:
+                    return BusinessProcessResult.Ok(request.TransactionId);
```

Probably a silly question but is this a sensible default? If the type to create isn't Consumption or Production then we just don't validate and say OK. Could we throw an error instead saying that this type isn't supported yet?

---

It's because we haven't implemented a domain model for other metering point types yet. Can add a TODO maybe also?